### PR TITLE
fix old usage of satori/uuid

### LINKED ISSUE #890: Incorrect import of "github.com/satori/uuid"
the correct import is github.com/satori/go.uuid

list files with incorrect import:
```
/azure-sdk-for-go$ egrep "satori/uuid" -r ./
./glide.lock:- name: github.com/satori/uuid
./storage/entity_test.go:	"github.com/satori/uuid"
./storage/table_batch.go:	"github.com/satori/uuid"
./storage/entity.go:	"github.com/satori/uuid"
./storage/table_batch_test.go:	"github.com/satori/uuid"
```

this leads to the issue that you have to setup alias for github.com/satori/uuid in glide file when azure-sdk-fo-go got imported thru 3rd package. 

Example:
if you want to use https://github.com/kubernetes/kubernetes/tree/master/pkg/cloudprovider/providers/azure package,
you would have to add duplicated entries to your glide.yalm 

```
- package: github.com/satori/go.uuid
  version: 1.1.0
- package: github.com/satori/uuid
  version: 1.1.0  
  vcs: git
  repo: https://github.com/satori/go.uuid.git
```

### PR DESCRIPTION
**Pull request by joshgav**
fixes #890

### CODE REVIEW

**Comment by tamird** [comment]
This fixed #890, but introduces a new problem: it is now impossible for downstream to use a version of github.csatori/go.uuid more recent than this package's (1.1.0 at the time of writing). Unfortunately 1.1.0 does not include this accidental bug fix: https://github.com/satori/go.uuid/commit/0aa62d5.

cc @sdboyer in case he has some ideas.

**Comment by sdboyer** [comment]
@tamird sorry, i don't have time to dig for it myself right now - could you summarize for me what about this change makes that impossible?

**Comment by tamird** [comment]
@sdboyer the maintainers of go.uuid have released a new version, so this is fine now.
